packages/angular/common/src/directives/navigation/router-link-delegate.ts

```diff
@@ -31,12 +31,45 @@ export class RouterLinkDelegateDirective implements OnInit, OnChanges {
 
   ngOnInit(): void {
     this.updateTargetUrlAndHref();
+    this.updateTabindex();
   }
 
   ngOnChanges(): void {
     this.updateTargetUrlAndHref();
   }
 
+  /**
+   * The `tabindex` is set to `0` by default on the host element when
+   * the `routerLink` directive is used. This causes issues with Ionic
+   * components that wrap an `a` or `button` element, such as `ion-item`.
+   * See issue https://github.com/angular/angular/issues/28345
+   *
+   * This method removes the `tabindex` attribute from the host element
+   * to allow the Ionic component to manage the focus state correctly.
+   */
+  private updateTabindex() {
+    // Ionic components that render a native anchor or button element
+    const ionicComponents = [
+      'ION-BACK-BUTTON',
```

Comment:
How should I test this one? It doesn't display when replacing `defaultHref` with `routerLink`.

Reply:
For `lazy`:

1. Update `packages/angular/test/base/src/app/lazy/router-link-page/router-link-page.component.html` to include `routerLink` on `ion-back-button`:
    ```html
    <ion-back-button routerLink="/lazy/router-link-page"></ion-back-button>
    ```
1. Rebuild / sync the app
1. Navigate to http://localhost:4200/lazy/router-link
1. Click on the first button `ion-button[routerLink]`
1. Attempt to tab to the `ion-back-button`
1. With my fix you will tab once to it, without it you will see a blue focus ring and have to double tab

For `standalone` you can add `routerLink` onto the back button at `packages/angular/test/base/src/app/standalone/back-button/back-button.component.html` and then navigate to http://localhost:4200/standalone/back-button but it doesn't seem to have `tabindex="0"` with or without the fix.